File: preshot_prediction/dependencies/ace_loggers/test/test_datalogger_demo.py
#!/usr/bin/env python3
"""
Confidential, Copyright 2024, Sony AI, All rights reserved.
"""

import contextlib
import os
import sys

import numpy
import pytest
from ace_loggers import datalogger_demo_pybind as datalogger


def test_ineritance():
    file_name = "test_ineritance.ace"
    data_str = "some text string!"

    with contextlib.suppress(FileNotFoundError):
        os.remove(file_name)

    with datalogger.DataWriter(file_name) as writer:
        writer.write(data_str)

    with datalogger.DataReader(file_name) as reader:
        assert data_str == reader.read_str()


def test_primitive():
    file_name = "test_primitive.ace"
    data_struct = datalogger.HasPrimitivesStruct(numpy.random.randint(0, 1000), numpy.random.rand())

    with contextlib.suppress(FileNotFoundError):
        os.remove(file_name)
    with datalogger.DataWriter(file_name) as writer:
        writer.write(data_struct)

    with datalogger.DataReader(file_name) as reader:
        loaded_struct = datalogger.HasPrimitivesStruct()
        reader.read(loaded_struct)
        assert data_struct.var_int == loaded_struct.var_int
        assert data_struct.var_float == pytest.approx(loaded_struct.var_float)


def test_overriding():
    file_name = "test_overriding.ace"
    data_struct = datalogger.HasPointerStruct(numpy.random.randint(0, 1000), numpy.random.rand(), "random text!")

    with contextlib.suppress(FileNotFoundError):
        os.remove(file_name)
    with datalogger.DataWriter(file_name) as writer:
        writer.write(data_struct)

    with datalogger.DataReader(file_name) as reader:
        loaded_struct = datalogger.HasPointerStruct()
        reader.read(loaded_struct)
        assert data_struct.var_int == loaded_struct.var_int
        assert data_struct.var_float == pytest.approx(loaded_struct.var_float)
        assert data_struct.var_str == loaded_struct.var_str


if __name__ == "__main__":
    sys.exit(pytest.main(sys.argv))

File: preshot_prediction/dependencies/ace_loggers/test/test_datalogger_demo.cpp
// Confidential, Copyright 2024, Sony AI, All rights reserved.
#include "ace_loggers/datalogger_demo.hpp"
#include "ace_loggers/datalogger_tests.hpp"

using namespace datalogger_demo;

TEST_F(DataLoggerTests, TestPrimitive) {
  const std::string file_name = "test.ace";
  const HasPrimitivesStruct write_data(UniformIntSample(0, 1000), NormalSample(0, 100));

  {
    std::remove(file_name.c_str());
    DataWriter writer(file_name);
    writer << write_data;
  }

  {
    DataReader reader(file_name);

    HasPrimitivesStruct read_data;
    reader >> read_data;
    EXPECT_EQ(write_data.var_int, read_data.var_int);
    EXPECT_EQ(write_data.var_float, read_data.var_float);
  }

  SUCCEED();
}

TEST_F(DataLoggerTests, TestOverriding) {
  const std::string file_name = "test.ace";
  const HasPointerStruct write_data(UniformIntSample(0, 1000), NormalSample(0, 100),
                                    RandomString(UniformIntSample(12, 32)));

  {
    std::remove(file_name.c_str());
    DataWriter writer(file_name);
    writer << write_data;
  }

  {
    HasPointerStruct read_data;
    {
      DataReader reader(file_name);
      reader >> read_data;
    }

    EXPECT_EQ(write_data.var_int, read_data.var_int);
    EXPECT_EQ(write_data.var_float, read_data.var_float);
    EXPECT_EQ(write_data.var_str, read_data.var_str);
  }

  SUCCEED();
}

int main(int argc, char* argv[]) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}
